### PR TITLE
Add first Nailgun tests

### PR DESCRIPTION
**Pull request by Duhemm**

### CODE REVIEW

File: frontend/src/test/scala/bloop/nailgun/NailgunTest.scala [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    /**
+     * Executes a command `cmd` expecting a 0 exit code.
+     *
+     * @param cmd The command to run.
+     */
+    def success(cmd: String*): Unit = {
+      val failMessage: String =
+        s"""Success expected, but command failed. Log were:
```
Comment (jvican) [inline frontend/src/test/scala/bloop/nailgun/NailgunTest.scala]: s/Log/Logs/g

File: frontend/src/main/scala/bloop/Server.scala [hunk 2]
```diff
@@ -51,6 +51,6 @@ object Server {
     import bloop.engine.State
     import bloop.engine.tasks.Tasks
     State.stateCache.allStates.foreach(s => Tasks.persist(s))
-    server.shutdown(true)
+    server.shutdown( /* exitVM = */ false)
```
Comment (jvican) [inline frontend/src/main/scala/bloop/Server.scala]: Why is this necessary?

File: .drone.yml [hunk 3]
```diff
@@ -45,7 +45,7 @@ pipeline:
       - /drone/.sbt
 
   build:
-    image: scalacenter/scala-docs:1.0
+    image: scalacenter/scala-docs:1.1
```
Comment (jvican) [inline .drone.yml]: I assume this new image has Nailgun or its deps inside? Python?

File: frontend/src/test/scala/bloop/nailgun/BasicNailgunSpec.scala [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    }
+  }
+
+  @Test
+  def compileCommandTest(): Unit = {
+    withServerInProject("with-resources") { (logger, client) =>
+      client.success("clean", "-p", "with-resources")
+      client.success("compile", "-p", "with-resources")
+      val messages = logger.getMessages()
+      val needle = "Compiling"
```
Comment (jvican) [inline frontend/src/test/scala/bloop/nailgun/BasicNailgunSpec.scala]: Can we check that "clean" also works? :smile:

File: project/BuildPlugin.scala [hunk 5]
```diff
@@ -71,6 +71,7 @@ object BuildKeys {
   val buildIntegrationsIndex =
     Def.taskKey[File]("A csv index with complete information about our integrations.")
   val buildIntegrationsBase = Def.settingKey[File]("The base directory for our integration builds.")
+  val nailgunClientLocation = Def.settingKey[sbt.File]("Where to find the python nailgun client")
```
Comment (jvican) [inline project/BuildPlugin.scala]: Neat.

File: frontend/src/test/scala/bloop/nailgun/NailgunTest.scala [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        .directory(base.toFile)
+    }
+
+    /**
+     * Executes a command `cmd` on the server, and return the exit code.
+     *
+     * @param cmd The command to execute
+     * @return The exit code of the operation.
+     */
+    def issue(cmd: String*): Int = {
```
Comment (jvican) [inline frontend/src/test/scala/bloop/nailgun/NailgunTest.scala]: Yeah, looks like there's no other way of doing it.

File: frontend/src/main/scala/bloop/Server.scala (hunk 2 above)
Comment (Duhemm) [reply frontend/src/main/scala/bloop/Server.scala]: Because we shutdown the server at the end of the test, and we don't want to exit the JVM that sbt nicely forked for us.

File: .drone.yml (hunk 3 above)
Comment (Duhemm) [reply .drone.yml]: It has only Python: See https://github.com/scalacenter/docker-scala-docs/pull/1

**Comment by Duhemm** [comment]
Comments addressed 🌮
